src/Faker/ORM/PDO/Populator.php

```diff
@@ -0,0 +1,173 @@
+        $columns = array();
+        $placeholders = array();
+
+        foreach ($this->rows[$tableKey][1] as $key => $value) {
+            $columns[] = $key;
+            $placeholders[] = ':' . $key;
+        }
+
+        $columns = implode(', ', $columns);
+        $placeholders = implode(', ', $placeholders);
+        $stmt = 'INSERT INTO '. trim($tableKey) .' (' . $columns . ') VALUES (' . $placeholders . ')';
+        $result = $this->connection->prepare($stmt);
+
+        return $result;
+    }
+
+    /**
+     * Convert the rows corresponding to the supplied table name to
+     * keys for use in pdo execute statement.
+     * @param  string $tableKey The table name this dataset is for.
+     * @return array            The array with placeholder keys
+     */
+    private function convertToPlaceholders($tableKey)
+    {
+        $originalArray = $this->rows[$tableKey];
+        $newArray = array();
+
+        for ($i=0; $i < count($originalArray); $i++) {
+            $newArray[$i] = array();
+            foreach ($originalArray[$i] as $key => $value) {
+                $newArray[$i][':'. $key] = $value;
+            }
+        }
+
+        return $newArray;
+    }
+
+    /**
+     * Return a faker value from the correct formatter.
+     * @param  string/array $request Data from supplied mapping
+     * @return string                Value returned by formatter or null
+     */
+    private function generateValue($request)
+    {
+        $final = false;
+        $reqType = gettype($request);
+        $count = count($request);
+
+        if ($reqType === 'array' && $count > 0) {
+            $name = trim($request[0]);
+
+            if ($count === 1) {
+                $params = array();
+            } else {
+                $params = $request[1];
+            }
+
+            // Run method
+            switch (count($params)) {
+                case 0:
+                    $final = $this->generator->$name();
+                    break;
+                case 1:
+                    $final = $this->generator->$name($params[0]);
+                    break;
+                case 2:
+                    $final = $this->generator->$name($params[0], $params[1]);
+                    break;
+                case 3:
+                    $final = $this->generator->$name($params[0], $params[1], $params[1]);
+                    break;
+                case 4:
+                    $final = $this->generator->$name($params[0], $params[1], $params[1], $params[3]);
+                    break;
+                case 5:
+                    $final = $this->generator->$name($params[0], $params[1], $params[1], $params[3], $params[4]);
+                    break;
+                case 6:
+                    $final = $this->generator->$name($params[0], $params[1], $params[1], $params[3], $params[4], $params[5]);
+                    break;
+            }
+        } else if ($reqType == 'string') {
+            $name = trim($request);
```

Comment:
what's `$name`? Impossible to tell by reading your code... Please find a better name.